components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/util/Utils.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.context.rewrite.util;
+
+import org.apache.catalina.connector.Request;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+public class Utils {
+
+    public static String getTenantDomainFromURLMapping(Request request) {
+        String requestURI = request.getRequestURI();
+        String domain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+
+        if (requestURI.contains("/t/")) {
+            String temp = requestURI.substring(requestURI.indexOf("/t/") + 3);
+            if (temp.indexOf('/') != -1) {
```

Comment:
assign the index to local variable and reuse in line `temp = temp.substring(0, temp.indexOf('/'));`